lib/bundler/cli/remove.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+module Bundler
+  class CLI::Remove
+    def initialize(gems)
+      @gems = gems
+      @removed_deps = []
+    end
+
+    def run
+      builder = Dsl.new
+      builder.eval_gemfile(Bundler.default_gemfile)
+
+      # store removed gems to display on successfull removal
+      @removed_deps = builder.remove_gems(@gems)
+
+      # remove gems from lockfile
+      @definition = builder.to_definition(Bundler.default_lockfile, {})
+      @definition.lock(Bundler.default_lockfile)
+
+      # remove those gems from the gemfile
+      remove_gems_from_gemfile
+
+      # TODO: Discuss about using this,
+      # currently gems are not removed from .bundle
+      # Installer.install(Bundler.root, Bundler.definition)
+
+      display_removed_gems
+    end
+
+  private
+
+    def remove_gems_from_gemfile
```

Comment:
does this actually assert that all requested gems _can_ be removed by the regular expressions?

Reply:
Not yet. Just the basic ones. It does not handle the nested groups well. Working on that for now.